rs/xnet/payload_builder/src/certified_slice_pool.rs

```diff
@@ -1159,6 +1159,17 @@ impl CertifiedSlicePool {
             None => return Ok(None),
         };
 
+        let signal_limit = get_signal_limit(
+            slice.payload.header.begin(),
+            slice
+                .payload
+                .messages
+                .as_ref()
+                .map(|messages| messages.begin()),
```

Comment:
Actually I just moved this whole computation further down into [take_prefix on Payload](https://sourcegraph.com/github.com/dfinity/ic@c0b2ec6580bdaf80047b9e7b125a86eab424adff/-/blob/rs/xnet/payload_builder/src/certified_slice_pool.rs?L380). Since it's now defined on the slice itself (`Payload` is part of `UnpackedStreamSlice`) it should be ok to use just `slice.begin` and `slice.header.begin` right? I mean if it's an invalid slice this would seem to be the wrong place to reject it anyway.